jsonargparse/_actions.py

```diff
@@ -699,6 +699,25 @@ def get_subcommands(
             if subcommand is None and not (fail_no_subcommand and action._required):  # type: ignore
                 return None, None
             if action._required and subcommand not in action._name_parser_map:  # type: ignore
+                if dest == "subcommand":
+                    # If this action is a subcommand action and no subcommand is provided,
```

Comment:
This if condition and comment do not make sense. If you look at line 670, in this context `action` is always a subcommand action. This is also the reason why there isn't 100% test coverage. The original `KeyError` is never being used.

Reply:
Revised, please check